transaction/core/src/ring_signature/mlsag.rs

```diff
@@ -332,24 +356,56 @@ fn challenge(
 }
 
 fn decompress_ring(
-    ring: &[(CompressedRistrettoPublic, CompressedCommitment)],
-) -> Result<Vec<(RistrettoPublic, Commitment)>, Error> {
+    ring: &[ReducedTxOut],
+) -> Result<Vec<(RistrettoPublic, Commitment)>, MLSAGError> {
     // Ring must decompress.
     let mut decompressed_ring: Vec<(RistrettoPublic, Commitment)> = Vec::new();
-    for (compressed_address, compressed_commitment) in ring {
-        let ristretto_public =
-            RistrettoPublic::try_from(compressed_address).map_err(|_e| Error::InvalidCurvePoint)?;
-        let commitment = Commitment::try_from(compressed_commitment)?;
+    for tx_out in ring {
+        let ristretto_public = RistrettoPublic::try_from(&tx_out.target_key)
+            .map_err(|_e| MLSAGError::InvalidCurvePoint)?;
+        let commitment = Commitment::try_from(&tx_out.commitment)?;
         decompressed_ring.push((ristretto_public, commitment));
     }
     Ok(decompressed_ring)
 }
 
+/// An error which can occur when signing or verifying
+#[derive(Clone, Debug, Deserialize, Display, Eq, Hash, Ord, PartialEq, PartialOrd, Serialize)]
+pub enum MLSAGError {
```

Comment:
I think we should not do this, because it's more consistent with all the existing code to use `MLSAG`. For example we have `RingMLSAG` in transaction-core, and in external facing protos. If you want to rename to `Mlsag`, let's discuss that in a separate PR and do it everywhere at once for consistency.

Also, if we do split this into another crate as planned, the struct will just be called `Error` at that point.